### PR TITLE
{bio}[foss/2021b] Trinity v2.15.1, BLAT v3.7, RSEM v1.3.3 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
@boegel please take a look at the test failures

**Comment by akesandgren** [comment]
HDF5 1.10.8 is used in ncbi-vdb due to ABI ioncompat from 1.12.x and that is the sole cause of the HDF5 problem.
So we probably need a whitelisting of that.
As for GATK we can just drop usage here to 4.2.3.0. Only DROP have it as a dep before this one.

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ('picard', '2.25.1', '-Java-%(javaver)s', SYSTEM),
+    ('GATK', '4.2.6.1', '-Java-%(javaver)s'),
+    ('Perl', '5.34.0'),
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('BLAST+', '2.12.0'),
+    ('BLAT', '3.7'),
+    ('Bowtie', '1.3.1'),
+    ('Bowtie2', '2.4.4'),
+    ('GMAP-GSNAP', '2021-21-17'),
```
Comment (akesandgren) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2021b.eb:48]: Incorrect version once #18231 has been merged. Please update

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2021b.eb (hunk 1 above)
Comment (Micket) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2021b.eb:48]:
```suggestion
    ('GMAP-GSNAP', '2021-12-17'),
```

**Comment by boegel** [comment]
Hmm, it's way trickier than just adding an exception in the tests...

`HISAT2` depends on `ncbi-vdb` 2.11.2 which in turns depends on HDF5 1.10.8, while `R` depends on HDF5 1.12.1.
Both `HISAT2` and `R` are dependencies for `RSEM`.

So the only way out is to change `ncbi-vdb-2.11.2-gompi-2021b.eb` so it uses HDF5 1.12.1 as well, and make it play nice with other easyconfigs from the same generation, like we did for `ncbi-vdb-3.0.0-gompi-2021b.eb` in #17140 ...

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 10 out of 10 (3 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/ec9933db0d42042e6651ee0957c10efb for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3169.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/ecd23a8b149ebc207acdf1a9a0008a11 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/akesandgren/25cb02e0b3202bce19f43b19c00c8eed for a full test report.

**Review by akesandgren — CHANGES_REQUESTED** [review]
LGTM

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**